### PR TITLE
Clear the waterfall label overlay on silent slots (#202)

### LINKED ISSUE #202: FT2/FT4 waterfall shows stale decodes that don't disappear
@
## Issue
Previously decoded messages remain displayed on the waterfall when they should disappear/clear.

## Evidence
- "same thing with repeating last decoded messages on waterfall"
- "it is not disappearing"

## Notes
Appears related to stale decode cleanup — decodes from prior cycles are not being cleared. May overlap with the per-cycle decode clearing logic.
@

### PR DESCRIPTION
**Pull request by patrickrb**
﻿Closes #202.

## Root cause
The waterfall stamps decoded-message labels onto its scrolling bitmap once per slot (`WaterfallLabelGate`). The label source — `MainViewModel.currentMessages` — was only updated when a slot produced decodes: `afterDecode` returns early on a **silent** slot (no decodes, or everything filtered out as own-TX echoes) **without** touching `currentMessages`. So every silent slot re-stamped the **previous** slot's messages, which appeared to repeat down the waterfall and never disappear. Worst on FT4 (7.5 s) and especially FT2 (3.8 s), whose short, often-empty slots re-stamp far more often than FT8's 15 s slots.

## Fix
Refresh the overlay on every pass via a new, unit-tested `WaterfallLabelMessages.afterPass(previous, kept, isDeep)`:
- a **normal** (non-deep) pass is authoritative for the slot — its result, even empty, becomes the overlay, so a silent slot clears the prior labels;
- a **deep** pass only augments — an empty deep pass keeps the normal pass's labels rather than wiping them.

`afterDecode` now routes all three `currentMessages` updates (no-decode early return, all-filtered early return, success path) through this helper.

## Test
`WaterfallLabelMessagesTest`: normal-empty (clears), normal-nonempty (replaces), deep-empty (keeps previous), deep-nonempty (replaces), and null-previous edge cases. `gradlew testDebugUnitTest --tests WaterfallLabelMessagesTest` passes.

> ⚠️ Unit-verified. On-air confirmation recommended: FT2/FT4 decodes should scroll off and not repeat once their slot is over.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: ft8cn/app/src/main/java/com/bg7yoz/ft8cn/ui/WaterfallLabelMessages.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+     * @return the overlay message list for the waterfall
+     */
+    public static ArrayList<Ft8Message> afterPass(ArrayList<Ft8Message> previous,
+                                                  ArrayList<Ft8Message> kept,
+                                                  boolean isDeep) {
+        if (isDeep && kept.isEmpty()) {
+            return previous;
+        }
+        return kept;
+    }
```
[the remark this answers is not in the document]
Comment (patrickrb) [reply ft8cn/app/src/main/java/com/bg7yoz/ft8cn/ui/WaterfallLabelMessages.java:63]: Good catch — confirmed. `addMsgToList` mutates `msgs` in place to strip anything already in `allMsg`, and `afterDecode` is handed that deduplicated `msgs`, so a deep pass really does carry only its *new* decodes. Returning `kept` therefore dropped the normal-pass labels. Fixed in 3fca7247: `afterPass` now merges `previous + kept` for a non-empty deep pass (null-previous safe), keeps `previous` for an empty deep pass, and still returns `kept` for normal passes — matching how `ft8Messages`/`currentDecodeCount` already accumulate across deep passes.

File: ft8cn/app/src/test/java/com/bg7yoz/ft8cn/ui/WaterfallLabelMessagesTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    @Test
+    public void deepPass_nonEmpty_replacesWithTheDeepResult() {
+        ArrayList<Ft8Message> previous = listOf(1);
+        ArrayList<Ft8Message> kept = listOf(4); // deep pass is a superset of the normal one
+        ArrayList<Ft8Message> result =
+                WaterfallLabelMessages.afterPass(previous, kept, true);
+        assertThat(result).isSameInstanceAs(kept);
+    }
```
[the remark this answers is not in the document]
Comment (patrickrb) [reply ft8cn/app/src/test/java/com/bg7yoz/ft8cn/ui/WaterfallLabelMessagesTest.java:82]: Updated in 3fca7247: this test now asserts the merged behavior (size 5, contains both `previous` and `kept`) instead of replacement, and I added a `deepPass_nonEmpty_withNullPrevious` case to cover the null-previous path.
